sys/walltime/shell.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (C) 2025 ML!PA Consulting GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     sys_walltime
+ * @{
+ *
+ * @file
+ * @brief       Wall-Clock time shell commands
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+#include <stdlib.h>
+#include <string.h>
+
+#include "shell.h"
+#include "rtc_utils.h"
+#include "walltime.h"
+
+static int cmd_uptime(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    uint32_t uptime_full = walltime_uptime(true);
+    uint32_t uptime_boot = walltime_uptime(false);
+
+    printf("%"PRIu32" s", uptime_boot);
+    if (uptime_boot != uptime_full) {
+        printf(" (%"PRIu32" s with hibernation)", uptime_full);
+    }
+    puts("");
+
+    return 0;
+}
+
+static int _print_time(struct tm *time)
```

Comment:
I have also copied this function too many times. There could be a `fmt_time_tm()` in the `fmt` module function.